Source/Details/ASCollectionElement.h

```diff
@@ -21,12 +23,14 @@ AS_SUBCLASSING_RESTRICTED
 //TODO change this to be a generic "kind" or "elementKind" that exposes `nil` for row kind
 @property (nonatomic, readonly, copy, nullable) NSString *supplementaryElementKind;
 @property (nonatomic, assign) ASSizeRange constrainedSize;
-@property (weak, nonatomic) id<ASTraitEnvironment> traitEnvironment;
+@property (nonatomic, weak, nullable) ASDisplayNode *owningNode;
+@property (nonatomic, assign) ASPrimitiveTraitCollection primitiveTraitCollection;
```

Comment:
`weak` implies `nullable` and let's name the property `traitCollection` because it's shorter.